src/System.Private.DataContractSerialization/src/System/Xml/XmlDictionaryString.cs

```diff
@@ -26,7 +26,7 @@ public class XmlDictionaryString
         public XmlDictionaryString(IXmlDictionary dictionary, string value, int key)
         {
             if (dictionary == null)
-                _dictionary = new EmptyStringDictionary();
+                throw System.Runtime.Serialization.DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentNullException("dictionary"));
```

Comment:
As far as I understand we want to have this to have the same behavior on desktop but I'm not aware of any related fix.